src/ucp/wireup/wireup_cm.c

```diff
@@ -1068,8 +1095,33 @@ void ucp_cm_server_conn_request_cb(uct_listener_h listener, void *arg,
                      UCT_DEVICE_NAME_MAX);
     memcpy(ucp_conn_request->remote_dev_addr, remote_data->dev_addr,
            remote_data->dev_addr_length);
+
+    /* [ sa_data | ucp_address | user_data ]
+     * |      conn_priv_data_length        |
+     * 1. sa_data is fixed of size, so copy it directly */
     memcpy(&ucp_conn_request->sa_data, remote_data->conn_priv_data,
-           remote_data->conn_priv_data_length);
+           sizeof(ucp_conn_request->sa_data));
+
+    /* 2. Unpack ucp_address and get it's length in buffer */
+    addr_flags                    = ucp_cm_address_pack_flags(worker);
+    ucp_conn_request->remote_addr = ucs_malloc(sizeof(ucp_unpacked_address_t),
+                                               "ucp_unpacked_address_t");
+
+    status = ucp_address_unpack(ucp_conn_request->listener->worker,
+                                UCS_PTR_BYTE_OFFSET(remote_data->conn_priv_data,
```

Comment:
maybe use ucs_serialize macros

Reply:
`remote_data->conn_priv_data` is a const pointer, would need to make non-const pointer to it to pass to `ucs_serialize_next`. This does just pointer arithmetic without changing pointer.